.github/workflows/dependabot-automerge.yml

```diff
@@ -0,0 +1,45 @@
+name: Dependabot Auto-Merge
+
+on:
+  pull_request:
+    branches: 
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Install GitHub CLI
+        run: sudo apt update && sudo apt install -y gh
+      
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v2
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+      
+      - name: Approve pull request
+        run: gh pr review --approve "$PR_URL"
+        env:
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Enable auto-merge for Dependabot PRs
+        run: |
+          UPDATE_TYPE="${{ steps.metadata.outputs.update-type }}"
+          DEPENDENCY_TYPE="${{ steps.metadata.outputs.dependency-type }}"
+          
+          if [[ "$UPDATE_TYPE" == "version-update:semver-minor" || "$UPDATE_TYPE" == "version-update:semver-patch" ]] && \
+             [[ "$DEPENDENCY_TYPE" == "direct-development" || "$DEPENDENCY_TYPE" == "direct" ]]; then
+            gh pr merge --merge "$PR_URL"
+          else
+            echo "PR does not meet auto-merge criteria"
```

Comment:
I see this line about the `auto-merge criteria`. Do you mind adding some comments in the `yml` file about what those are?